python/ray/data/tests/test_object_gc.py

```diff
@@ -126,26 +126,22 @@ def test_iter_batches_no_spilling_upon_shuffle(shutdown_only):
 
 def test_pipeline_splitting_has_no_spilling(shutdown_only):
     # The object store is about 800MiB.
-    ctx = ray.init(num_cpus=1, object_store_memory=800e6)
+    ctx = ray.init(num_cpus=1, object_store_memory=1200e6)
     # The size of dataset is 50000*(80*80*4)*8B, about 10GiB, 50MiB/block.
-    ds = ray.data.range_tensor(50000, shape=(80, 80, 4), parallelism=200)
+    ds = ray.data.range_tensor(5000, shape=(80, 80, 4), parallelism=20)
 
     # 2 blocks/window.
-    p = ds.window(bytes_per_window=100 * 1024 * 1024).repeat()
+    p = ds.window(bytes_per_window=100 * 1024 * 1024).repeat(2)
     p1, p2 = p.split(2)
 
     @ray.remote
     def consume(p):
         for batch in p.iter_batches(batch_size=None):
             pass
+        print(p.stats())
 
     tasks = [consume.remote(p1), consume.remote(p2)]
-    try:
-        # Run it for 20 seconds.
-        ray.get(tasks, timeout=20)
-    except Exception:
-        for t in tasks:
-            ray.cancel(t, force=True)
+    ray.get(tasks)
```

Comment:
@jianoaix FYI I found these tests can sometimes spuriously pass with the timeout, if the first iteration never starts. I think it's safer to force a fixed number of iterations (e.g., 2 here), which would be enough to trigger spilling if GC wasn't working correctly.

Reply:
Sounds good.